lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
The 1st case references `self` while the 2nd one keeps pointers to `self.cache` and `self.ids`. If `self.cache` or `self.ids` has not been reassigned, the result should be the same.